app.js

```diff
@@ -186,3 +190,21 @@ function syncToTarget (event, target) {
     server.replicateNetwork(socket, 'pull')
   }
 }
+
+function getGlobalDatasetCentroid (done) {
+  var bbox = [[-90,90],[-180,180]]
+
+  var stream = osm.queryStream(bbox)
+  var ix = setInterval(function () {
```

Comment:
You are right. I think my advice might have been bad. Perhaps the way to do this is with through2 and `stream.unpipe()` when we get the lat/lon? I think your concern is valid - when you added the `on('data')` listener it would have been put in streaming mode and would have continued to take processor cycles even after stopping reading.